testing/testing_index.adoc

```diff
@@ -0,0 +1,97 @@
+// vim: set syntax=asciidoc:
+[[Testing]]
+== Testing
+:data-uri:
+:toc:
+:toclevels 4:
+:homepage https://github.com/projectatomic/container-best-practices:
+
+=== Purpose of the testing
+
+Container images generally consist of distro packages and some scripts that help to start the container properly. For example in case of MariaDB database, it is set of RPMs with mariadb-server that provides the main functionality, but we need some scripts that handle proper database stack initialization, changes basic configuration based on the container run options etc.
+
+If we want to test basic functionality of the container, we do not have to test the RPM functionality, because it should have already be done during RPM develpment process. We need to focus on testing the additional scripts instead and test the API of the container -- whether it works as it is described. Again, in case of the MariaDB database, we do not run MariaDB unit test in the container, but we focus on whether the database is initialized properly, whether configuration is changed after proper options are set on container run command etc.
+
+=== Converntions for test scripts
+
+It is a good practice to keep the basic sanity test for the image together with the image sources. For example test/run might be a script, that tests image specified by IMAGE_NAME environment variable.
+
+=== Examples of test scripts
+
+Some examples of test scripts might be found in the container images for OpenShift, for example:
```

Comment:
These examples of test scripts can be found in the container images for OpenShift: